### ISSUE TITLE
Set a key from cli if it doesn't exist

### ISSUE THREAD

**Issue by clofresh** [hashicorp/consul#3052]
### `consul version` for both Client and Server
Client: `v0.7.2`
Server: `v0.7.2`

### `consul info` for both Client and Server
Client & Server:
```
agent:                                                                                                                                                                                                                     [7/1858]
        check_monitors = 0
        check_ttls = 0
        checks = 2
        services = 3
build:
        prerelease =
        revision = 'a9afa0c
        version = 0.7.2
consul:
        bootstrap = true
        known_datacenters = 1
        leader = true
        leader_addr =  <addr>
        server = true
raft:
        applied_index = 71220
        commit_index = 71220
        fsm_pending = 0
        last_contact = never
        last_log_index = 71220
        last_log_term = 2
        last_snapshot_index = 65539
        last_snapshot_term = 2
        latest_configuration = [{Suffrage:Voter ID:<addr> Address:<addr>}]
        latest_configuration_index = 1
        num_peers = 0
        protocol_version = 1
        protocol_version_max = 3
        protocol_version_min = 0
        snapshot_version_max = 1
        snapshot_version_min = 0
        state = Leader
        term = 2
runtime:
        arch = amd64
        cpu_count = 8
        goroutines = 56
        max_procs = 8
        os = linux
        version = go1.7.3
serf_lan:
        encrypted = false
        event_queue = 1
        event_time = 2
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 1
        members = 1
        query_queue = 0
        query_time = 1
serf_wan:
        encrypted = false
        event_queue = 0
        event_time = 1
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 1
        members = 1
        query_queue = 0
        query_time = 1
```

### Operating system and Environment details

ubuntu 14.04

### Description of the Issue (and unexpected/desired result)

```
$ consul kv put -cas -modify-index=0 some-new-key foo
Must specify -modify-index with -cas!
```

I was expected this key to get set because it doesn't exist yet, but I think the cli is using 0 as "null" here. The [kv docs](https://www.consul.io/api/kv.html#create-update-key) say:

```
If the index is 0, Consul will only put the key if it does not already exist.
```

**Comment by jjshanks** [comment]
I don't think this is an issue anymore, at least on the head of master.

```
❯ ./bin/consul version
Consul v1.9.0-dev
Revision 52451cf84+CHANGES
Protocol 2 spoken by default, understands 2 to 3 (agent will automatically use protocol >2 when speaking to compatible agents)
❯ ./bin/consul kv put -cas -modify-index=0 some-new-key foo
Success! Data written to: some-new-key
```

**Comment by blake** [comment]
That's correct. This was resolved by #8331 and will be in Consul 1.9.
